### DISCUSSION TITLE
Not getting errors through

### DISCUSSION THREAD

**Discussion by jim-at-jibba** [TanStack/query#391] [General] [answered]
Hey, 

Thanks for the awesome package. 

I am sure I am missing something but when I make a request with `useQuery` and it fails the error object is always empty. But checking in the network tab shows the error is returned. I would assume this would be returned.

Thanks

**Comment by Haqverdi** [comment]
Hi, i am also have same issue, are you resolved this?

**Comment by tannerlinsley** [comment, marked as answer]
For React Query to catch errors, they must be thrown inside your query function. If you are using `fetch` for example, it doesn't throw on a network error by default.

**Reply by jim-at-jibba** [reply]
That makes sense. Thanks

**Reply by orest22** [reply]
I would also like to add that it seems as it depends on the "retry" option error gets evaluated after all retries were made. @tannerlinsley please correct me if I am wrong it just based on my own observation.

**Reply by Wellers0n** [reply]
mock inside test 
```js
const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retryDelay: 1,
        retry: 0,
      },
    },
  });
```

**Reply by goors** [reply]
You would expect error right away with `retryDelay: 1`, or `retry: 0`, works for me.
